File: .sdk-repos/version-8.9/orchestration-cluster-api-js/examples/role.ts
// Compilable usage examples for role management operations.
// These examples are type-checked during build to guard against API regressions.

import { createCamundaClient, type Username } from '@camunda8/orchestration-cluster-api';

//#region CreateRole
async function createRoleExample() {
  const camunda = createCamundaClient();

  const result = await camunda.createRole({
    roleId: 'process-admin',
    name: 'Process Admin',
  });

  console.log(`Created role: ${result.roleId}`);
}
//#endregion CreateRole

//#region GetRole
async function getRoleExample() {
  const camunda = createCamundaClient();

  const role = await camunda.getRole(
    { roleId: 'process-admin' },
    { consistency: { waitUpToMs: 5000 } }
  );

  console.log(`Role: ${role.name}`);
}
//#endregion GetRole

//#region SearchRoles
async function searchRolesExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchRoles(
    {
      page: { limit: 10 },
    },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const role of result.items ?? []) {
    console.log(`${role.roleId}: ${role.name}`);
  }
}
//#endregion SearchRoles

//#region UpdateRole
async function updateRoleExample() {
  const camunda = createCamundaClient();

  await camunda.updateRole({
    roleId: 'process-admin',
    name: 'Process Administrator',
  });
}
//#endregion UpdateRole

//#region DeleteRole
async function deleteRoleExample() {
  const camunda = createCamundaClient();

  await camunda.deleteRole({ roleId: 'process-admin' });
}
//#endregion DeleteRole

//#region AssignRoleToUser
async function assignRoleToUserExample(username: Username) {
  const camunda = createCamundaClient();

  await camunda.assignRoleToUser({
    roleId: 'process-admin',
    username,
  });
}
//#endregion AssignRoleToUser

//#region UnassignRoleFromUser
async function unassignRoleFromUserExample(username: Username) {
  const camunda = createCamundaClient();

  await camunda.unassignRoleFromUser({
    roleId: 'process-admin',
    username,
  });
}
//#endregion UnassignRoleFromUser

//#region AssignRoleToGroup
async function assignRoleToGroupExample() {
  const camunda = createCamundaClient();

  await camunda.assignRoleToGroup({
    roleId: 'process-admin',
    groupId: 'engineering-team',
  });
}
//#endregion AssignRoleToGroup

//#region UnassignRoleFromGroup
async function unassignRoleFromGroupExample() {
  const camunda = createCamundaClient();

  await camunda.unassignRoleFromGroup({
    roleId: 'process-admin',
    groupId: 'engineering-team',
  });
}
//#endregion UnassignRoleFromGroup

//#region AssignRoleToClient
async function assignRoleToClientExample() {
  const camunda = createCamundaClient();

  await camunda.assignRoleToClient({
    roleId: 'process-admin',
    clientId: 'my-service-account',
  });
}
//#endregion AssignRoleToClient

//#region UnassignRoleFromClient
async function unassignRoleFromClientExample() {
  const camunda = createCamundaClient();

  await camunda.unassignRoleFromClient({
    roleId: 'process-admin',
    clientId: 'my-service-account',
  });
}
//#endregion UnassignRoleFromClient

//#region AssignRoleToMappingRule
async function assignRoleToMappingRuleExample() {
  const camunda = createCamundaClient();

  await camunda.assignRoleToMappingRule({
    roleId: 'process-admin',
    mappingRuleId: 'rule-123',
  });
}
//#endregion AssignRoleToMappingRule

//#region UnassignRoleFromMappingRule
async function unassignRoleFromMappingRuleExample() {
  const camunda = createCamundaClient();

  await camunda.unassignRoleFromMappingRule({
    roleId: 'process-admin',
    mappingRuleId: 'rule-123',
  });
}
//#endregion UnassignRoleFromMappingRule

//#region SearchUsersForRole
async function searchUsersForRoleExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchUsersForRole(
    { roleId: 'process-admin' },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const user of result.items ?? []) {
    console.log(`User: ${user.username}`);
  }
}
//#endregion SearchUsersForRole

//#region SearchGroupsForRole
async function searchGroupsForRoleExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchGroupsForRole(
    { roleId: 'process-admin' },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const group of result.items ?? []) {
    console.log(`Group: ${group.groupId}`);
  }
}
//#endregion SearchGroupsForRole

//#region SearchClientsForRole
async function searchClientsForRoleExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchClientsForRole(
    { roleId: 'process-admin' },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const client of result.items ?? []) {
    console.log(`Client: ${client.clientId}`);
  }
}
//#endregion SearchClientsForRole

//#region SearchMappingRulesForRole
async function searchMappingRulesForRoleExample() {
  const camunda = createCamundaClient();

  const result = await camunda.searchMappingRulesForRole(
    { roleId: 'process-admin' },
    { consistency: { waitUpToMs: 5000 } }
  );

  for (const rule of result.items ?? []) {
    console.log(`Mapping rule: ${rule.name}`);
  }
}
//#endregion SearchMappingRulesForRole

// Suppress "declared but never read"
void createRoleExample;
void getRoleExample;
void searchRolesExample;
void updateRoleExample;
void deleteRoleExample;
void assignRoleToUserExample;
void unassignRoleFromUserExample;
void assignRoleToGroupExample;
void unassignRoleFromGroupExample;
void assignRoleToClientExample;
void unassignRoleFromClientExample;
void assignRoleToMappingRuleExample;
void unassignRoleFromMappingRuleExample;
void searchUsersForRoleExample;
void searchGroupsForRoleExample;
void searchClientsForRoleExample;
void searchMappingRulesForRoleExample;

// Suppress "declared but never read"
void createRoleExample;
void getRoleExample;
void searchRolesExample;
void updateRoleExample;
void deleteRoleExample;
void assignRoleToUserExample;
void unassignRoleFromUserExample;
void assignRoleToGroupExample;
void unassignRoleFromGroupExample;
void assignRoleToClientExample;
void unassignRoleFromClientExample;
void assignRoleToMappingRuleExample;
void unassignRoleFromMappingRuleExample;
void searchUsersForRoleExample;
void searchGroupsForRoleExample;
void searchClientsForRoleExample;
void searchMappingRulesForRoleExample;

File: .sdk-repos/orchestration-cluster-api-go/examples/role.go
// Role operations: CRUD plus membership assignment (users, groups, clients,
// mapping rules).
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func createRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region CreateRole
	result, err := client.CreateRole(ctx, *openapi.NewRoleCreateRequest("auditor", "Auditor"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion CreateRole
	return nil
}

func searchRolesExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchRoles
	result, err := client.SearchRoles(ctx, *openapi.NewRoleSearchQueryRequest())
	if err != nil {
		return err
	}
	for _, r := range result.GetItems() {
		fmt.Printf("%v\n", r)
	}
	// endregion SearchRoles
	return nil
}

func getRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetRole
	role, err := client.GetRole(ctx, "auditor")
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", role)
	// endregion GetRole
	return nil
}

func updateRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region UpdateRole
	result, err := client.UpdateRole(ctx, "auditor", *openapi.NewRoleUpdateRequest("Senior Auditor"))
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion UpdateRole
	return nil
}

func deleteRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region DeleteRole
	return client.DeleteRole(ctx, "auditor")
	// endregion DeleteRole
}

func searchUsersForRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchUsersForRole
	result, err := client.SearchUsersForRole(ctx, "auditor", *openapi.NewRoleUserSearchQueryRequest())
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion SearchUsersForRole
	return nil
}

func assignRoleToUserExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region AssignRoleToUser
	return client.AssignRoleToUser(ctx, "auditor", "alice")
	// endregion AssignRoleToUser
}

func unassignRoleFromUserExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region UnassignRoleFromUser
	return client.UnassignRoleFromUser(ctx, "auditor", "alice")
	// endregion UnassignRoleFromUser
}

func searchGroupsForRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchGroupsForRole
	result, err := client.SearchGroupsForRole(ctx, "auditor", *openapi.NewRoleGroupSearchQueryRequest())
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion SearchGroupsForRole
	return nil
}

func assignRoleToGroupExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region AssignRoleToGroup
	return client.AssignRoleToGroup(ctx, "auditor", "finance")
	// endregion AssignRoleToGroup
}

func unassignRoleFromGroupExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region UnassignRoleFromGroup
	return client.UnassignRoleFromGroup(ctx, "auditor", "finance")
	// endregion UnassignRoleFromGroup
}

func searchClientsForRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchClientsForRole
	result, err := client.SearchClientsForRole(ctx, "auditor", *openapi.NewRoleClientSearchQueryRequest())
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion SearchClientsForRole
	return nil
}

func assignRoleToClientExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region AssignRoleToClient
	return client.AssignRoleToClient(ctx, "auditor", "reporting-service")
	// endregion AssignRoleToClient
}

func unassignRoleFromClientExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region UnassignRoleFromClient
	return client.UnassignRoleFromClient(ctx, "auditor", "reporting-service")
	// endregion UnassignRoleFromClient
}

func searchMappingRulesForRoleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region SearchMappingRulesForRole
	result, err := client.SearchMappingRulesForRole(ctx, "auditor", *openapi.NewMappingRuleSearchQueryRequest())
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion SearchMappingRulesForRole
	return nil
}

func assignRoleToMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region AssignRoleToMappingRule
	return client.AssignRoleToMappingRule(ctx, "auditor", "sso-auditors")
	// endregion AssignRoleToMappingRule
}

func unassignRoleFromMappingRuleExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region UnassignRoleFromMappingRule
	return client.UnassignRoleFromMappingRule(ctx, "auditor", "sso-auditors")
	// endregion UnassignRoleFromMappingRule
}
